### PR TITLE
fix: hide file tabs on legacy challenges

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
This is based off https://github.com/freeCodeCamp/freeCodeCamp/pull/43967, hence the pile of commits.

With SHOW_UPCOMING_CHANGES true:

Before:

Desktop
image

Mobile
image

After: 

Desktop
image
Mobile
image

### CODE REVIEW

**Comment by ShaunSHamilton** [comment]
Did you create this branch from the wrong branch...

**Comment by ojeytonwilliams** [comment]
> Did you create this branch from the wrong branch...

> This is based off #43967, hence the pile of commits.

Nope, I didn't want to deal with merge conflicts.

**Review by naomi-lgbt — DISMISSED** [review]
This looks good to me.

File: client/src/templates/Challenges/classic/desktop-layout.tsx [hunk 1]
```diff
@@ -77,18 +78,25 @@ const DesktopLayout = (props: DesktopLayoutProps): JSX.Element => {
...
-  const isPreviewDisplayable = projectBasedChallenge
+  const displayPreview = projectBasedChallenge
     ? showPreview && hasPreview
     : hasPreview;
-  const isConsoleDisplayable = projectBasedChallenge ? showConsole : true;
+  const displayConsole = projectBasedChallenge ? showConsole : true;
   const { codePane, editorPane, instructionPane, previewPane, testsPane } =
     layoutState;
 
+  const displayEditorTabs =
```
Comment (moT01) [inline client/src/templates/Challenges/classic/desktop-layout.tsx:94]: Why not just `usesMultifileEditor` here? In which case, you could just use that as the check below.

I gave it a quick test, you do need at least a check for `!hasEditableBoudaries` or the tabs show up twice. But, I'm wondering, why the editor tabs are ever needed on the desktop view - wouldn't they always be in the `ActionRow` when on desktop?

File: client/src/templates/Challenges/classic/mobile-layout.tsx [hunk 2]
```diff
@@ -49,14 +50,20 @@ class MobileLayout extends Component<MobileLayoutProps> {
...
 
     const editorTabPaneProps = {
       mountOnEnter: true,
       unmountOnExit: true
     };
 
+    // This is not the same logic as the desktop version since the mobile
+    // version does not have an ActionRow, but does need a way to switch between
+    // the different tabs.
+    const displayEditorTabs = showUpcomingChanges && usesMultifileEditor;
```
Comment (moT01) [inline client/src/templates/Challenges/classic/mobile-layout.tsx:65]: Why not just `usesMultiFileEditor` here?

**Review by moT01 — COMMENTED** [review]
I would be fine adding this as-is, it all works - but I left a few comments below.

File: client/src/templates/Challenges/classic/desktop-layout.tsx (hunk 1 above)
Comment (ojeytonwilliams) [reply client/src/templates/Challenges/classic/desktop-layout.tsx:94]: Overabundance of caution, to be honest.  I wanted to be sure the tabs were definitely hidden until we removed the `SHOW_UPCOMING_CHANGES` flag.

I'm fine with totally removing them for the desktop, though.

File: client/src/templates/Challenges/classic/mobile-layout.tsx [hunk 3]
```diff
@@ -49,14 +50,20 @@ class MobileLayout extends Component<MobileLayoutProps> {
...
     } = this.props;
 
     const editorTabPaneProps = {
       mountOnEnter: true,
       unmountOnExit: true
     };
 
+    // This is not the same logic as the desktop version since the mobile
+    // version does not have an ActionRow, but does need a way to switch between
+    // the different tabs.
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/classic/mobile-layout.tsx:64]:
```suggestion
    // Unlike the desktop layout the mobile version does not have an ActionRow,
    // but still needs a way to switch between the different tabs.
```

File: client/src/templates/Challenges/classic/mobile-layout.tsx (hunk 2 above)
Comment (ojeytonwilliams) [reply client/src/templates/Challenges/classic/mobile-layout.tsx:64]: We could, but this is a (minor) way to make sure we don't accidentally show the tabs until we're ready.

**Comment by ShaunSHamilton** [comment]
Maybe dumb question: Has the JS console always been visible for legacy (soon-to-be) challenges?
image


I noticed, because this piece of logic seemed incorrect for RWD stuff (`true` when not `projectBased`)
```js
const displayConsole = projectBasedChallenge ? showConsole : true;
```

**Comment by moT01** [comment]
Pretty sure it's always been like that - but I'm not positive. Test output goes there.

**Review by moT01 — DISMISSED** [review]
LGTM 🎉

**Comment by nhcarrigan** [comment]
image

Leaving a note here that I lose the editor tabs on mobile.

**Comment by nhcarrigan** [comment]
So the tabs appear on the branch as is, but not when rebased against `main`.

**Comment by ojeytonwilliams** [comment]
> So the tabs appear on the branch as is, but not when rebased against main.

Should be fixed now.  The metas needed updating.

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍

**Review by naomi-lgbt — APPROVED** [review]
Looks good to me!

**Pull request merged**